### PR TITLE
transmission update to 4.0.5

### PR DESCRIPTION
**Pull request by HouLiXieBuRou**
Topic Description
-----------------

- transmission: update to 4.0.5
Enable support gtkmm-4 and libdelfate

- gtkmm-4: new,4.0.2
To introduce transmission

- pangomm-2.48: new,2.50.1
To introduce gtkmm-4

- cairomm-1.16: new,1.18.0
To introduce Pangomm-2.48

Package(s) Affected
-------------------

Security Update?
----------------

No

Build Order
-----------

```
#buildit cairomm-1.16 pangomm-2.48 gtkmm-4 transmission
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by HouLiXieBuRou** [comment]
amd64 test
image
arm64 test
image

**Comment by HouLiXieBuRou** [comment]
amd64 gui test
image

File: desktop-gnome/gtkmm-4/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=gtkmm-4
+PKGSEC=gnome
+PKGDEP="gtk-4 pangomm-2.48 atkmm"
+BUILDDEP="doxygen graphviz libxslt mm-common"
+PKGDES="C++ bindings for GTK+4"
```
Comment (MingcongBai) [inline desktop-gnome/gtkmm-4/autobuild/defines]: GTK 4 (GTK+ is no longer used):

C++ bindings for GTK 4

File: desktop-gnome/gtkmm-4/spec [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VER=4.0.2
+SRCS="tbl::https://download.gnome.org/sources/gtkmm/${VER:0:3}/gtkmm-$VER.tar.xz"
+CHKSUMS="sha256::0c836e8daffd836ef469499b7a733afda3a5260ea0e4d81c552f688ae384bd97"
+CHKUPDATE="html::url=https://download.gnome.org/sources/gtkmm/${VER:0:4}/;pattern=gtkmm-(.+?).tar.xz"
```
Comment (MingcongBai) [inline desktop-gnome/gtkmm-4/spec]: Use anitya.

File: runtime-desktop/pangomm-2.48/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+BUILDDEP__LOONGSON2F="${BUILDDEP__RETRO}"
+BUILDDEP__POWERPC="${BUILDDEP__RETRO}"
+BUILDDEP__PPC64="${BUILDDEP__RETRO}"
+
+ABTYPE=meson
+MESON_AFTER="-Dbuild-documentation=true"
+MESON_AFTER__RETRO=" \
+             ${MESON_AFTER} \
+             -Dbuild-documentation=false"
+MESON_AFTER__ARMV4="${MESON_AFTER__RETRO}"
```
Comment (MingcongBai) [inline runtime-desktop/pangomm-2.48/autobuild/defines]: Lines 21-28 is not necessary.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**